src/lib/realtime.js

```diff
@@ -9,30 +9,29 @@ const APPS_DOCTYPE = 'io.cozy.apps'
  * @param {object}
  * @returns {Promise}
  */
-async function initializeRealtime({ getApp, onCreate, onDelete, url, token }) {
-  const realtimeConfig = { token, url }
-
+async function initializeRealtime({ getApp, onCreate, onDelete, cozyClient }) {
   try {
-    realtime
-      .subscribe(realtimeConfig, APPS_DOCTYPE)
-      .onCreate(async app => {
-        // Fetch directly the app to get attributes `related` as well.
-        let fullApp
-        try {
-          fullApp = await getApp(app.slug)
-        } catch (error) {
-          throw new Error(`Cannot fetch app ${app.slug}: ${error.message}`)
-        }
+    const realtime = new CozyRealtime({ cozyClient })
+
+    realtime.subscribe('created', APPS_DOCTYPE, async app => {
+      // Fetch directly the app to get attributes `related` as well.
+      let fullApp
+      try {
+        fullApp = await getApp(app.slug)
+      } catch (error) {
+        throw new Error(`Cannot fetch app ${app.slug}: ${error.message}`)
+      }
+
+      if (typeof onCreate === 'function') {
+        onCreate(fullApp)
+      }
+    })
 
-        if (typeof onCreate === 'function') {
-          onCreate(fullApp)
-        }
-      })
-      .onDelete(app => {
-        if (typeof onDelete === 'function') {
-          onDelete(app)
-        }
-      })
+    realtime.subscribe('deleted', APPS_DOCTYPE, app => {
+      if (typeof onDelete === 'function') {
+        onDelete(app)
+      }
+    })
```

Comment:
I do not see any unsubscribe, is it OK ?

Reply:
There was none before, so I do not really know where to put it

Reply:
We don't handle unmounting within the CozyBar so I don't think we have a place to unsubscribe realtime :/

Reply:
I tried to do something here https://github.com/cozy/cozy-bar/pull/548